tutorials/tour/_posts/2017-02-13-nested-functions.md

```diff
@@ -11,25 +11,27 @@ next-page: currying
 previous-page: higher-order-functions
 ---
 
-In Scala it is possible to nest function definitions. The following object provides a `filter` function for extracting values from a list of integers that are below a threshold value:
+In Scala it is possible to nest function definitions. The following object provides a `factorial` function for computing the factorial of a given number:
 
 ```tut
-object FilterTest extends App {
-  def filter(xs: List[Int], threshold: Int) = {
-    def process(ys: List[Int]): List[Int] =
-      if (ys.isEmpty) ys
-      else if (ys.head < threshold) ys.head :: process(ys.tail)
-      else process(ys.tail)
-    process(xs)
+object FactorialTest extends App {
```

Comment:
I think it's clearer if all of the examples are in worksheet mode and not inside classes.